opm/simulators/wells/MultisegmentWell_impl.hpp

```diff
@@ -1394,7 +1399,12 @@ namespace Opm
                 break;
             }
 
-            residual_history.push_back(this->getWellResiduals(Base::B_avg_, deferred_logger));
+            // getFinteWellResiduals returns false for nan/inf residuals
+            const auto& [isFinite, residuals] = this->getFiniteWellResiduals(Base::B_avg_, deferred_logger);
```

Comment:
Unfortunately, this produces warnings:
```
MultisegmentWell_impl.hpp:1403:26: warning: declaration of ‘isFinite’ shadows a previous local [-Wshadow]
 1403 |             const auto& [isFinite, residuals] = this->getFiniteWellResiduals(Base::B_avg_, deferred_logger);
      |                          ^~~~~~~~
```
Maybe we should rename the variables here to avoid this? `const auto& [isFinite2, residuals2]` or use curly brackets to give each of these it's own scope?